### PR TITLE
Check message integrity.

### PR DESCRIPTION
**Pull request by frodsan**

### CODE REVIEW

File: lib/facebook/messenger/server.rb [hunk 1]
```diff
@@ -36,6 +37,8 @@ def verify_token
       end
 
       def receive
+        return if app_secret && !check_integrity
```
Comment (frodsan) [inline lib/facebook/messenger/server.rb]: If integrity check fails, the endpoint returns 200. Not sure if this is ok, but I think the integrity check will only fail if a request doesn't come from Facebook. If you return 50x, Facebook will retry the request many times.

Also, It only does integrity check if `app_secret` is present.

**Comment by jgorset** [comment]
Oh, cool – I didn't know this was a thing! Where'd you find it in the documentation?

**Comment by frodsan** [comment]
I checked the headers and follow GitHub's guide: https://developer.github.com/webhooks/securing/ 😂 

I just found this: https://developers.facebook.com/docs/graph-api/webhooks#receiveupdates

**Comment by frodsan** [comment]
I can add tests and remove the `rewind` thingy if this looks good to merge. Later, you can decide what to respond in case that integrity fails, but probably it doesn't matter if an attacker receives a 200. I need to investigate more.

**Comment by jgorset** [comment]
Cool! Looks good to me, so please do.

**Comment by frodsan** [comment]
Done!

**Comment by gauravtiwari** [comment]
this is pretty and good 😉

**Comment by jgorset** [comment]
:cake:

**Pull request merged**